devito/types/dense.py

```diff
@@ -960,7 +962,23 @@ def __indices_setup__(cls, **kwargs):
                 raise TypeError("Need either `grid` or `dimensions`")
         elif dimensions is None:
             dimensions = grid.dimensions
-        return dimensions
+
+        # Staggered indices
+        staggered = kwargs.get("staggered", None)
+        if staggered in [CELL, NODE]:
+            staggered_indices = dimensions
```

Comment:
these aren't really staggered are they?

Reply:
No they are, but are located on the node rather than half-node. Staggered refers to two interleaving grids shifted by half-point so you have bot on the node and half off the node. So if everything is on the NODE, then yes you are back to cartesian but if there is any field shifted then FD needs to know it is a staggered grid object but located on the NODE.